src/app/pages/update-profile/update-profile.page.ts

```diff
@@ -0,0 +1,137 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {City} from "../../models/City";
+import {AlertController, ModalController} from "@ionic/angular";
+import {Profile} from "../../models/Profile";
+import { Camera, CameraOptions } from '@ionic-native/camera/ngx';
+import {SearchCityPage} from "../search-city/search-city.page";
+import {RunningLevelPage} from "../running-level/running-level.page";
+
+@Component({
+  selector: 'app-update-profile',
+  templateUrl: './update-profile.page.html',
+  styleUrls: ['./update-profile.page.scss'],
+})
+export class UpdateProfilePage implements OnInit {
+
+  @Input() profile: Profile;
+  public customActionSheetOptions: object = {
+    header: 'Selectionnez votre sexe',
+  };
+  public choiceBirthday: string;
+  public profileUpdated: Profile;
+
+  constructor(private modalController: ModalController, private camera: Camera, private alertController: AlertController) { }
+
+  public dismissModalUpdateProfile(): void {
+    this.modalController.dismiss();
+  }
+
+  public async presentAlertCamera(cover: boolean): Promise<void> {
+    const alert = await this.alertController.create({
+      header : (cover) ? 'Photo de couverture' : 'Photo de profile',
+      buttons: [
+        {
+          text: 'choisir une photo',
+          role: 'ok',
+          handler: () => {
+            this.addPhoto(cover, 'library');
+          }
+        },
+        {
+          text: 'Prendre une photo',
+          handler: () => {
+            this.addPhoto(cover, 'camera');
+          }
+        }
+      ]
+    });
+
+    await alert.present();
+  }
+
+  public update(): void {
+    this.profile = this.profileUpdated;
+    this.profile.birthday = new Date(this.choiceBirthday).getTime();
+    console.log(this.profile)
+    this.dismissModalUpdateProfile();
+  }
+
+  public async modalSearchCity(): Promise<void> {
+    const modal = await this.modalController.create({
+      component: SearchCityPage,
+      swipeToClose: true
+    });
+    await modal.present();
+    const { data } = await modal.onWillDismiss();
+    this.profileUpdated.city = Object.assign(new City(), {
+      name: data.city.name,
+      code: data.city.code
+    });
+  }
+
+  public async modalRunningLevel(): Promise<void> {
+    const modal = await this.modalController.create({
+      component: RunningLevelPage,
+      swipeToClose: true
+    });
+    await modal.present();
+    const { data } = await modal.onWillDismiss();
+    this.profileUpdated.runningLevel = data.runningLevel;
+  }
+
+  private async addPhoto(cover: boolean, source: string): Promise<void> {
+    let image;
```

Comment:
picture ce serait mieux.

Reply:
Coder plus simplement...

`const picture: <type> = source === 'library' ? await this.openLibrary() : await this.openCamera();`
`this.profileUpdated.cover = 'data:image/jpg;base64,' : 'data:image/jpg;base65;' + image;`